frontend/src/components/namespace/CreateNamespaceButton.tsx

```diff
@@ -0,0 +1,126 @@
+
+  const dispatch = useDispatch();
+  const kubeNamespace = Namespace;
+
+  function toggleCreateNamespaceVisibility() {
+    setNamespaceDialogOpen(namespaceDialogOpen => !namespaceDialogOpen);
+  }
+
+  console.log('kubeNamespace', kubeNamespace);
+
+  function createNewNamespace() {
+    const newNamespaceData = {
+      apiVersion: 'v1',
+      kind: 'Namespace',
+      metadata: {
+        name: namespaceName,
+      },
+    };
+
+    // need this for the item name?
+    const itemName = newNamespaceData.metadata.name;
+
+    // need a callback so just made one to use
+    const callback = (itemName: string) => {
+      console.log(`Callback called with ${itemName}`);
+    };
+
+    async function createNamesapce(newNamespaceData: any) {
+      try {
+        const response = await post('/api/v1/namespaces', newNamespaceData);
+        console.log('response', response);
+
+        toggleCreateNamespaceVisibility();
+        dispatch(
+          clusterAction(callback.bind(itemName), {
+            startMessage: t('Creating namespace {{ itemName }}…', { itemName }),
+            cancelledMessage: t('Cancelled creation of {{ itemName }}.', { itemName }),
+            successMessage: t('Created namesapce {{ itemName }}.', { itemName }),
+            errorMessage: t('Error creating namespace {{ itemName }}.', { itemName }),
+            cancelUrl: location.pathname,
+            // startUrl: item!.getListLink(),
+            // errorUrl: item!.getListLink(),
+            // ...options,
+          })
+        );
+
+        return response;
+      } catch (error) {
+        console.error('Error creating namespace', error);
+
+        toggleCreateNamespaceVisibility();
+        return error;
+      }
+    }
+
+    createNamesapce(newNamespaceData);
+  }
+
+  return (
+    <AuthVisible item={kubeNamespace} authVerb="create" namespace="default">
+      <ActionButton
+        color="primary"
+        description={t('translation|Create')}
+        icon={'mdi:plus-circle'}
+        onClick={() => {
+          toggleCreateNamespaceVisibility();
+        }}
+      />
+
+      <Dialog open={namespaceDialogOpen} onClose={() => toggleCreateNamespaceVisibility()}>
+        <DialogTitle>{t('translation|Create Namespace')}</DialogTitle>
+        <DialogContent>
+          <Box component="form">
+            <Typography>Please enter the name of the new namespace</Typography>
+            <TextField
+              margin="dense"
+              id="name"
+              label={t('glossary|Namespace')}
+              type="text"
+              fullWidth
+              value={namespaceName}
+              onChange={event => setNamespaceName(event.target.value)}
+            />
+          </Box>
+        </DialogContent>
+        <DialogActions>
+          <Button
+            onClick={() => {
+              toggleCreateNamespaceVisibility();
+              dispatchCreateEvent({
+                status: EventStatus.CONFIRMED,
+              });
```

Comment:
This is reporting that the namespace creation was confirmed when in fact this button cancels it.